include/ode_solvers/leapfrog.hpp

```diff
@@ -71,31 +124,52 @@ struct LeapfrogODESolver {
 
       // x <- x + eta v'
       Point y = xs[v_index];
-      y = (eta) * y;
 
       if (Ks[x_index] == NULL) {
-        xs[x_index] = xs_prev[x_index] + y;
+        xs[x_index] = xs_prev[x_index] + eta*y;
       }
       else {
         // Find intersection (assuming a line trajectory) between x and y
-        do {
-
-          pbpair = Ks[x_index]->line_positive_intersect(xs_prev[x_index], y, Ar, Av);
-
-          if (pbpair.first >= 0 && pbpair.first <= 1) {
+        bool step_completed = false;
+        NT T = eta;
+        if (k == 0 && !accepted) {
```

Comment:
variable `k` corresponds to the walk length? If yes, then why `k==0` in this condition? If (k!=0 and !accepted) holds then the initialization in line 136 is again necessary, right?

Reply:
See below.